proto/kvrpcpb.proto

```diff
@@ -24,23 +24,10 @@ message LockInfo {
     optional bytes  key          = 3;
 }
 
-// ResultType use for many different result, so dispose each Type prudently.
-message ResultType {
-    enum Type {
-        Ok          = 1;
-        Retryable   = 2;
-        Locked      = 3;
-        Committed   = 4;
-        Rolledback  = 5;
-        NotLeader   = 6;
-        // Known result type add here.
-        Other       = 9;
-    }
-    optional Type type                          = 1 [default = Other];
-    optional string msg                         = 2;
-    // If and only if Type == Locked
-    optional LockInfo lock_info                 = 3;
-    optional errorpb.NotLeader leader_info      = 4;
+message KeyError {
+    optional LockInfo locked    = 1; // Client should backoff or cleanup the lock then retry.
+    optional string   retryable = 2; // Client may restart the txn. e.g write conflict.
```

Comment:
@siddontang For simplifying handle logic in TiClient.
TiClient does not really care about what causes the error, it just logs the error message then return a retryable error to TiDB.